### PR TITLE
[JENKINS-32503] Update dependency of beanutils

### PR DESCRIPTION
**Pull request by KostyaSha**
https://issues.jenkins-ci.org/browse/JENKINS-32503

Testing PR build with dependency update as is.
Jenkins is already on 1.8.3 https://github.com/jenkinsci/jenkins/blob/74c534207ca0b97b0345bf413b63f2450ccc8d04/core/pom.xml#L277-L279
That fails 

---

This change is [Reviewable](https://reviewable.io/reviews/stapler/stapler/81)

### CODE REVIEW

**Comment by KostyaSha** [comment]
No PR builds? 0_o

**Comment by KostyaSha** [comment]
https://jenkins.ci.cloudbees.com/job/core/job/stapler/58/
Only one failure :)

**Comment by KostyaSha** [comment]
Commit that introduced that test https://github.com/stapler/stapler/commit/c3004deca59f842614f54203a33bdb2d89ce10d5

**Comment by KostyaSha** [comment]
Any help is welcome ;)

**Comment by KostyaSha** [comment]
This test also fails on 1.8.3 that used in Jenkins.

**Comment by KostyaSha** [comment]
@rsandell maybe you can help? :)

File: core/src/test/java/org/kohsuke/stapler/DataBindingTest.java [hunk 1]
```diff
@@ -398,4 +400,34 @@ public void testDerivedProperty() {
...
+        public FluentSetter() {}
+
+        /**
+         * WIP This tests passes in stapler, but fluent doesn't work in jenkins...
+         * @see FluentPropertyBeanIntrospector
+         *
+         * It is also possible to transform it to `withXXX()` to have clear vision.
+         */
+        @DataBoundSetter
+//        public void setItems(List<String> items) {
```
Comment (oleg-nenashev) [inline core/src/test/java/org/kohsuke/stapler/DataBindingTest.java]: dead code

File: core/src/test/java/org/kohsuke/stapler/DataBindingTest.java [hunk 2]
```diff
@@ -398,4 +400,34 @@ public void testDerivedProperty() {
...
     }
+
+    public static class FluentSetter {
+        private List<String> items;
+
+        @DataBoundConstructor
+        public FluentSetter() {}
+
+        /**
+         * WIP This tests passes in stapler, but fluent doesn't work in jenkins...
```
Comment (oleg-nenashev) [inline core/src/test/java/org/kohsuke/stapler/DataBindingTest.java]: CC @vivek

File: core/src/test/java/org/kohsuke/stapler/DataBindingTest.java (hunk 1 above)
Comment (KostyaSha) [inline core/src/test/java/org/kohsuke/stapler/DataBindingTest.java]: Of course it dead code, because it WIP

**Comment by jglick** [comment]
`DataBindingTest` failing.

**Comment by KostyaSha** [comment]
> DataBindingTest failing.

it's known, i have no power to fix scalarTest and seems i'm missing something...

**Comment by oleg-nenashev** [comment]
@KostyaSha  could you please rebase it to the master

File: core/pom.xml [hunk 3]
```diff
@@ -44,7 +44,7 @@
     <dependency>
       <groupId>commons-beanutils</groupId>
       <artifactId>commons-beanutils</artifactId>
-      <version>1.7.0</version>
+      <version>1.9.3</version>
```
Comment (jglick) [inline core/pom.xml]: Too new, and see #134.

File: core/src/test/java/org/kohsuke/stapler/DataBindingTest.java [hunk 4]
```diff
@@ -412,4 +414,34 @@ public void testDerivedProperty() {
...
+         * @see FluentPropertyBeanIntrospector
+         *
+         * It is also possible to transform it to `withXXX()` to have clear vision.
+         */
+        @DataBoundSetter
+//        public void setItems(List<String> items) {
+        public FluentSetter setItems(List<String> items) {
```
Comment (jglick) [inline core/src/test/java/org/kohsuke/stapler/DataBindingTest.java]: Seems potentially useful, but why is it filed here? Should be split into a separate PR.

**Comment by KostyaSha** [comment]
When we tried with oleg it seems passed test. Kohsuke changed databinding
in core, so maybe it already works without beanutils update. Need try
again.

File: core/src/test/java/org/kohsuke/stapler/DataBindingTest.java (hunk 4 above)
Comment (KostyaSha) [reply core/src/test/java/org/kohsuke/stapler/DataBindingTest.java]: Fluentsetters was exactly the reason to update beanutils that provides fluentbinder used in core... But i had working testcase here and failing in core + unclear test failure stuck me.

File: core/pom.xml (hunk 3 above)
Comment (johnypony3) [reply core/pom.xml]: Can this be revisited? We are running into some serious dependency issues because its at 1.8.3

**Comment by jglick** [comment]
Compare #188.

**Comment by KostyaSha** [comment]
@jglick so if it everything is solved now, then core and test could simplify code with using fluent setters. Requires changes to return `this` instead void.

**Comment by jglick** [comment]
Potentially useful new API, but would need various sorts of work to be mergeable.
